library/core/src/mem/maybe_uninit.rs

```diff
@@ -117,28 +117,22 @@ use crate::slice;
 /// `MaybeUninit<T>` can be used to initialize a large array element-by-element:
 ///
 /// ```
-/// use std::mem::{self, MaybeUninit};
+/// use std::mem::MaybeUninit;
 ///
 /// let data = {
-///     // Create an uninitialized array of `MaybeUninit`. The `assume_init` is
-///     // safe because the type we are claiming to have initialized here is a
-///     // bunch of `MaybeUninit`s, which do not require initialization.
-///     let mut data: [MaybeUninit<Vec<u32>>; 1000] = unsafe {
-///         MaybeUninit::uninit().assume_init()
-///     };
+///     // Create an uninitialized array of `MaybeUninit`.
+///     let mut data = [MaybeUninit::uninit(); 1000];
```

Comment:
This only works because the type is `Copy`, while the old example *deliberately* uses a non-Copy type. Please show how to init non-Copy arrays.

Reply:
Doesn't the `String` example right below show that?

Reply:
That doesn't initialize an entire array, it just works with a partially filled array. IMO we should have an example for actually crating a `[T; N]` for a non-Copy `T` via in-place init.

Reply:
Gotya, done.

Reply:
The `Copy` example is also outright silly, since one could just write `[42; 1000]`. So in its current from I think we'd be better off without that example. If you know an actually useful example with a Copy type, feel free to use that.

Reply:
Basically I think we'd be best off with exactly the old example, except that it uses the new transposition conversion. I'm open to improving the example but would like to know which issue is fixed by that improvement. :)

Reply:
Yup yup, done. :)